lua/nvim-treesitter/ts_utils.lua

```diff
@@ -2,35 +2,21 @@ local api = vim.api
 
 local parsers = require "nvim-treesitter.parsers"
 local utils = require "nvim-treesitter.utils"
+local ts_query = vim.treesitter.query
 
 local M = {}
 
 --- Gets the actual text content of a node
+-- @deprecated Use vim.treesitter.query.get_node_text
 -- @param node the node to get the text from
 -- @param bufnr the buffer containing the node
 -- @return list of lines of text of the node
 function M.get_node_text(node, bufnr)
-  local bufnr = bufnr or api.nvim_get_current_buf()
-  if not node then
-    return {}
-  end
-
-  -- We have to remember that end_col is end-exclusive
-  local start_row, start_col, end_row, end_col = M.get_node_range(node)
-
-  if start_row ~= end_row then
-    local lines = api.nvim_buf_get_lines(bufnr, start_row, end_row + 1, false)
-    lines[1] = string.sub(lines[1], start_col + 1)
-    -- end_row might be just after the last line. In this case the last line is not truncated.
-    if #lines == end_row - start_row + 1 then
-      lines[#lines] = string.sub(lines[#lines], 1, end_col)
-    end
-    return lines
-  else
-    local line = api.nvim_buf_get_lines(bufnr, start_row, start_row + 1, false)[1]
-    -- If line is nil then the line is empty
-    return line and { string.sub(line, start_col + 1, end_col) } or {}
-  end
+  require("nvim-treesitter.utils").notify(
+    "nvim-treesitter.utils is deprecated: use vim.treesitter.query.get_node_text",
```

Comment:
```suggestion
    "nvim-treesitter.ts_utils is deprecated: use vim.treesitter.query.get_node_text",
```

I don't think there was any decision of upstreaming  `ts_utils` _in its entirety_ so far?

Reply:
no, just this function

Reply:
Yes, so why write "nvim-treesitter.ts_utils`" (implying the whole module) "is deprecated"?